src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs

```diff
@@ -11,68 +9,63 @@ namespace UKHO.S100PermitService.Common.Services
 {
     public class ManufacturerKeyService : IManufacturerKeyService
     {
-        private readonly IOptions<ManufacturerKeyConfiguration> _manufacturerKeyVault;
         private readonly ILogger<ManufacturerKeyService> _logger;
         private readonly ICacheProvider _cacheProvider;
         private readonly ISecretClient _secretClient;
 
-        public ManufacturerKeyService(IOptions<ManufacturerKeyConfiguration> manufacturerKeyVault,
-                                        ILogger<ManufacturerKeyService> logger,
-                                        ICacheProvider cacheProvider,
-                                        ISecretClient secretClient)
+        public ManufacturerKeyService(ILogger<ManufacturerKeyService> logger,
+                                      ICacheProvider cacheProvider,
+                                      ISecretClient secretClient)
         {
-            _manufacturerKeyVault = manufacturerKeyVault ?? throw new ArgumentNullException(nameof(manufacturerKeyVault));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _cacheProvider = cacheProvider ?? throw new ArgumentNullException(nameof(cacheProvider));
             _secretClient = secretClient ?? throw new ArgumentNullException(nameof(secretClient));
-            CacheManufacturerKeys();
         }
 
         public void CacheManufacturerKeys()
         {
             _logger.LogInformation(EventIds.ManufacturerKeyCachingStart.ToEventId(), "Caching Of Manufacturer Keys started.");
-            if(!string.IsNullOrEmpty(_manufacturerKeyVault.Value.ServiceUri))
+
+            var secretProperties = _secretClient.GetPropertiesOfSecrets();
```

Comment:
Why are we doing this? The code should read on demand and cache accordingly?

Reply:
Removed